ucp/continuous_ucx_progress.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+# See file LICENSE for terms.
+
+import asyncio
+import socket
+import weakref
+
+
+class ProgressTask(object):
+    def __init__(self, ctx, event_loop):
+        """Creates a task that keeps calling ctx.progress()
+
+        Notice, class and created task is carefull not to hold a
+        reference to `ctx` so that a danling progress task will
+        not prevent `ctx` to be garbage collected.
+
+        Parameters
+        ----------
+        ctx: ApplicationContext
+            The application context to progress
+        event_loop: asyncio.EventLoop
+            The event loop to do progress in.
+        """
+        self.weakref_ctx = weakref.ref(ctx)
+        self.event_loop = event_loop
+        self.asyncio_task = None
+
+    def __del__(self):
+        if self.asyncio_task is not None:
+            self.asyncio_task.cancel()
```

Comment:
What happens if the task scopes out normally?

Reply:
That is fine, when this goes out of scope the continuous progress will stop

Reply:
Ok, am just wondering do we need to call `.cancel()` explicitly or does this already happens for us?

Reply:
In some cases we have to call it explicitly.
PS: note that it is fine to call `.cancel()` even if the future is already done.